src/SimplePDPService.sol

```diff
@@ -105,6 +105,28 @@ contract SimplePDPService is PDPListener, PDPRecordKeeper, Initializable, UUPSUp
         return 2880;
     }
 
+    // Number of epochs at the end of a provoing period during which a 
+    // proof of possession can be submitted
+    function challengeWindow() public pure returns (uint256) {
+        return 60;
+    }
+
+    // The start of the next challenge window
+    // Useful for querying before nextProvingPeriod to determine challengeEpoch
+    function nextChallengeWindowStart(uint256 setId) public view returns (uint256) {
+        if (provingDeadlines[setId] == 0) {
```

Comment:
Does this If apply only when the proofSet is created (i.e. provingDeadlines[setId] is set to 0 when the proofSet is created?

Reply:
Yes.  In solidity variables are initialized to their zero value.  There should be no way for the code to assign 0 to provingDeadlines[setId] after the first assignment.